### DISCUSSION TITLE
Adding new fields across all subtypes when converting ADTs?

### DISCUSSION THREAD

**Discussion by codesurf42** [scalalandio/chimney#498] [Ideas]
I am wondering if this is currently somehow possible in Chimney (I have seen coproductInstance but this is for a single subtype handling) without rewriting every single case of ADT?

```scala
sealed trait Foo
object Foo {
  case class Sub1(a: String)
  case class Sub2(a: String)
}

sealed trait Bar {
  def b: Int
}
object Bar {
  case class Sub1(a: String, b: Int)
  case class Sub2(a: String, b: Int)
}
def convert(in: Foo) = in.into[Bar].withFieldConst(_.b, 1).transform
```

I really like the full ADT conversions and that kind of extensions could make them more handy to use.

**Comment by MateuszKubuszok** [comment]
Hello,

With the batch of changes merged and released yesterday it is possible to have something like:

```scala
def convert(in: Foo) = in.into[Bar].withFieldConst(_.matching[Bar].b, 1).transform
```

(see [Scastie](https://scastie.scala-lang.org/1bNHZMg3RDq04kacW91S6Q)).

The `.matching[Bar]` is currently necessary due to how the new nested-paths-to-overriden values were implemented ("when going down the AST follow the type-matching path in config", here we provide a hack where AST rule matches this nested path for all subtypes), but it could be polished in the future.

**Reply by codesurf42** [reply]
Thank you, this looks really nice! 🚀

**Reply by codesurf42** [reply]
Being able to rewrite full ADTs with additional fields is a great addition to chimney. Thanks for all you work @MateuszKubuszok! I hope to find some time to get more involved with chimney development - or at least popularisation :)

**Reply by MateuszKubuszok** [reply]
It should work without `_.matching` since [1.0.0-RC1](https://github.com/scalalandio/chimney/releases/tag/1.0.0-RC1)
